src/zenml/orchestrators/step_runner.py

```diff
@@ -0,0 +1,283 @@
+            output_artifacts: The output artifacts of the step.
+            step_run_info: The step run info.
+
+        Raises:
+            StepInterfaceError: If the output signature of the step does not
+                match the actual output of the step.
+        """
+        from zenml.steps import BaseParameters
+
+        step_name = self.configuration.name
+        step_entrypoint = self._load_step_entrypoint()
+        output_materializers = self._load_output_materializers()
+
+        # Building the args for the entrypoint function
+        function_params = {}
+
+        # First, we parse the inputs, i.e., params and input artifacts.
+        spec = inspect.getfullargspec(inspect.unwrap(step_entrypoint))
+        args = spec.args
+
+        if args and args[0] == "self":
+            args.pop(0)
+
+        for arg in args:
+            arg_type = spec.annotations.get(arg, None)
+            arg_type = resolve_type_annotation(arg_type)
+
+            if issubclass(arg_type, BaseParameters):
+                step_params = arg_type.parse_obj(self.configuration.parameters)
+                function_params[arg] = step_params
+            elif issubclass(arg_type, StepContext):
+                context = arg_type(
+                    step_name=step_name,
+                    output_materializers=output_materializers,
+                    output_artifacts=output_artifacts,
+                )
+                function_params[arg] = context
+            else:
+                # At this point, it has to be an artifact, so we resolve
+                function_params[arg] = self._load_input_artifact(
+                    input_artifacts[arg], arg_type
+                )
+
+        # Wrap the execution of the step function in a step environment
+        # that the step function code can access to retrieve information about
+        # the pipeline runtime, such as the current step name and the current
+        # pipeline run ID
+        with StepEnvironment(
+            step_run_info=step_run_info,
+        ):
+            self._stack.prepare_step_run(info=step_run_info)
+            step_failed = False
+            try:
+                return_values = step_entrypoint(**function_params)
+            except:  # noqa: E722
+                step_failed = True
+                raise
+            finally:
+                self._stack.cleanup_step_run(
+                    info=step_run_info, step_failed=step_failed
+                )
+
+        output_annotations = parse_return_type_annotations(spec.annotations)
+        if len(output_annotations) > 0:
+            # if there is only one output annotation (either directly specified
+            # or contained in an `Output` tuple) we treat the step function
+            # return value as the return for that output
+            if len(output_annotations) == 1:
+                return_values = [return_values]
+            elif not isinstance(return_values, Sequence):
```

Comment:
Mhm this is old code from the TFX executor, but very interesting suggestion, I'll debug and see if this leads to any issues

Reply:
Nice catch, this would indeed have passed (since probably the intial 0.5.0 release):

```python
@step
def my_step() -> Output(a=str, b=str):
  return "ab"
```

The two outputs would have been "a" and "b". I prevented this and added a test for it.